components/sidebar/sidebar_category/sidebar_category.tsx

```diff
@@ -0,0 +1,189 @@
+
+import React from 'react';
+import {Tooltip} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+import classNames from 'classnames';
+
+import {CategoryTypes} from 'mattermost-redux/constants/channel_categories';
+import {Channel} from 'mattermost-redux/types/channels';
+import {ChannelCategory} from 'mattermost-redux/types/channel_categories';
+import {localizeMessage} from 'mattermost-redux/utils/i18n_utils';
+
+import {trackEvent} from 'actions/diagnostics_actions';
+import OverlayTrigger from 'components/overlay_trigger';
+import Constants, {A11yCustomEventTypes} from 'utils/constants';
+import {isKeyPressed} from 'utils/utils';
+
+import SidebarChannel from '../sidebar_channel';
+
+type Props = {
+    category: ChannelCategory;
+    channels: Channel[];
+    setChannelRef: (channelId: string, ref: HTMLLIElement) => void;
+    handleOpenMoreDirectChannelsModal: (e: Event) => void;
+    getChannelRef: (channelId: string) => HTMLLIElement | undefined;
+    isCollapsed: boolean;
+    actions: {
+        setCategoryCollapsed: (categoryId: string, collapsed: boolean) => void;
+    };
+};
+
+export default class SidebarCategory extends React.PureComponent<Props> {
+    categoryTitleRef: React.RefObject<HTMLButtonElement>;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.categoryTitleRef = React.createRef();
+    }
+
+    componentDidMount() {
+        // Refs can be null when this component is shallowly rendered for testing
+        if (this.categoryTitleRef.current) {
+            this.categoryTitleRef.current.addEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
+            this.categoryTitleRef.current.addEventListener(A11yCustomEventTypes.DEACTIVATE, this.handleA11yDeactivateEvent);
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.categoryTitleRef.current) {
+            this.categoryTitleRef.current.removeEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
+            this.categoryTitleRef.current.removeEventListener(A11yCustomEventTypes.DEACTIVATE, this.handleA11yDeactivateEvent);
+        }
+    }
+
+    handleA11yActivateEvent = () => {
+        if (this.categoryTitleRef.current) {
+            this.categoryTitleRef.current.addEventListener('keydown', this.handleA11yKeyDown);
+        }
+    }
+
+    handleA11yDeactivateEvent = () => {
+        if (this.categoryTitleRef.current) {
+            this.categoryTitleRef.current.removeEventListener('keydown', this.handleA11yKeyDown);
+        }
+    }
+
+    handleA11yKeyDown = (e: KeyboardEvent) => {
+        if (isKeyPressed(e, Constants.KeyCodes.ENTER)) {
+            this.handleCollapse();
+        }
+    }
+
+    renderChannel = (channel: Channel) => {
+        const {isCollapsed} = this.props;
+
+        return (
+            <SidebarChannel
+                channelId={channel.id}
+                setChannelRef={this.props.setChannelRef}
```

Comment:
really minor Nit: if we destructure part of `props` at top we might as well destructure all available props used in the func